src/ucs/datastruct/ptr_array.h

```diff
@@ -121,12 +145,226 @@ void *ucs_ptr_array_replace(ucs_ptr_array_t *ptr_array, unsigned index, void *ne
                     !__ucs_ptr_array_is_free(_var = (void*)((_ptr_array)->start[_index])))
 
 
+/**
+ * For-each user function: Calculates how many free elements are ahead.
+ *
+ * @param [in]  _elem        An element in the ptr array.
+ *
+ * @return _size_elem - The number of free elements ahead if free, if not 1.
+ */
+static inline uint32_t
+__ucs_ptr_array_for_each_get_step_size(ucs_ptr_array_elem_t elem)
```

Comment:
please accept void* parameter and do casting inside. This way you will not need to do casting every single call to this func

Reply:
btw, why need to call it `for_each`? You can use this func outside for_each as well

Reply:
We cannot call it from outside the for_each, since it's variable - The algorithm is based on the fact that we store the number of consecutive free elements ahead from each free list item inside the freelist element and then we can skip the free items.

Regarding the void * - I fixed it, thanks!